# Tarea Epidemiología:  Medición de la concordancia

---
## 0. Carga y limpieza de datos.

In [34]:
# Importamos las librerias necesarias.
import pandas as pd
import numpy as np

In [35]:
# Extraemos el dataset del documento (en Excel) en un dataframe.
df = pd.read_excel('base_datos_covid.xlsx')
df

Num  EDD SEXO                       OCUPACION  \
0      1   52    F                        Empleado   
1      2   48    F                        Empleado   
2      3   40    F                      Enfermeras   
3      4   40    M                        Empleado   
4      5   31    M  Otros Trabajadores de la Salud   
..   ...  ...  ...                             ...   
370  371   42    M               Otras Ocupaciones   
371  372   29    M               Otras Ocupaciones   
372  373   38    F                         Médicos   
373  374   45    F                      Enfermeras   
374  375   40    F                      Enfermeras   

                                             MUNICIPIO CONSULTORIO  \
0    TLALPAN                                                     7   
1    XOCHIMILCO                                                  3   
2    XOCHIMILCO                                                  9   
3    COYOACAN                                                    5   
4    XOCHIMILCO                                                 10   
..                                                 ...         ...   
370  VENUSTIANO CARRANZA                                        10   
371  NICOLAS ROMERO                                             20   
372  COYOACAN                                                    6   
373  XOCHIMILCO                                                  4   
374  XOCHIMILCO                                                  2   

          TURNO FECHA_ATENCION_INGRESO FECHA_INICIO_CUADRO_CLINICO  \
0    Vespertino             2020-12-31                  2020-12-22   
1    Vespertino             2021-01-04                  2021-01-02   
2    Vespertino             2021-01-14                  2021-01-05   
3      Matutino             2021-01-05                  2021-01-03   
4    Vespertino             2021-01-05                  2021-01-03   
..          ...                    ...                         ...   
370  Vespertino             2022-06-28                  2022-06-24   
371  Vespertino             2022-06-20                  2022-06-20   
372  Vespertino             2022-07-05                  2022-07-02   
373    Matutino             2022-07-01                  2022-07-01   
374    Matutino             2022-08-10                  2022-08-10   

     INICIO_SUBITO  ...  ANOSMIA  DISGEUSIA  OTROS  FECHA_MUESTRA_PCR  \
0                0  ...        1          1      0         2021-01-04   
1                0  ...        0          0      0         2021-01-10   
2                1  ...        0          0      0         2021-01-11   
3                0  ...        0          0      0         2021-01-16   
4                0  ...        1          1      0         2021-01-06   
..             ...  ...      ...        ...    ...                ...   
370              0  ...        0          0      0         2022-06-28   
371              1  ...        0          0      0         2022-06-23   
372              1  ...        0          0      0         2022-07-05   
373              0  ...        1          0      0         2022-07-08   
374              0  ...        0          0      0         2022-08-10   

         RESULTADO PCR  FECHA_RESULTADO_LAB   FECHA_TOMA_PRUEBA_RAPIDA  \
0             POSITIVO           08/01/2021  31/12/2020 12:00:00 a. m.   
1             POSITIVO           15/01/2021  04/01/2021 12:00:00 a. m.   
2             POSITIVO           14/01/2021  05/01/2021 12:00:00 a. m.   
3             NEGATIVO           20/01/2021  05/01/2021 12:00:00 a. m.   
4             NEGATIVO           09/01/2021  05/01/2021 12:00:00 a. m.   
..                 ...                  ...                        ...   
370           NEGATIVO           30/06/2022  28/06/2022 12:00:00 a. m.   
371           NEGATIVO           24/06/2022  20/06/2022 12:00:00 a. m.   
372           NEGATIVO           07/07/2022  05/07/2022 12:00:00 a. m.   
373  MUESTRA RECHAZADA           09/07/2022  01/07/2022 1

In [36]:
# Seleccionamos las columnas de mayor interes para la tarea.
selec_col = df[['Num','EDD','SEXO','MUNICIPIO','RESULTADO PCR', 'DESC_RESULT_PRUEBA_RAPIDA', 'INICIO_SUBITO', 
               'FIEBRE', 'TOS', 'CEFALEA', 'ODINOFAGIA', 'ATAQUE_AL_ESTADO_GENERAL', 'MIALGIAS', 'ARTRALGIAS', 
               'POSTRACION', 'RINORREA', 'ESCALOFRIO', 'DOLOR_ABDOMINAL', 'CONJUNTIVITIS', 'DISNEA', 'CIANOSIS', 
               'DIARREA', 'DOLOR_TORACICO', 'POLIPNEA', 'IRRITABILIDAD_MENOS5AÑOS', 'CORIZA', 'ANOSMIA', 'DISGEUSIA', 
               'OTROS']]
selec_col

Num  EDD SEXO                                          MUNICIPIO  \
0      1   52    F  TLALPAN                                             
1      2   48    F  XOCHIMILCO                                          
2      3   40    F  XOCHIMILCO                                          
3      4   40    M  COYOACAN                                            
4      5   31    M  XOCHIMILCO                                          
..   ...  ...  ...                                                ...   
370  371   42    M  VENUSTIANO CARRANZA                                 
371  372   29    M  NICOLAS ROMERO                                      
372  373   38    F  COYOACAN                                            
373  374   45    F  XOCHIMILCO                                          
374  375   40    F  XOCHIMILCO                                          

         RESULTADO PCR DESC_RESULT_PRUEBA_RAPIDA  INICIO_SUBITO  FIEBRE  TOS  \
0             POSITIVO                  NEGATIVO              0       0    1   
1             POSITIVO                  POSITIVO              0       1    1   
2             POSITIVO                  NEGATIVO              1       1    1   
3             NEGATIVO                  NEGATIVO              0       1    1   
4             NEGATIVO                  NEGATIVO              0       0    1   
..                 ...                       ...            ...     ...  ...   
370           NEGATIVO                  NEGATIVO              0       0    1   
371           NEGATIVO                  NEGATIVO              1       1    1   
372           NEGATIVO                  NEGATIVO              1       0    1   
373  MUESTRA RECHAZADA                  NEGATIVO              0       1    1   
374           POSITIVO                  NEGATIVO              0       0    1   

     CEFALEA  ...  DISNEA  CIANOSIS  DIARREA  DOLOR_TORACICO  POLIPNEA  \
0          1  ...       0         0        0               0         0   
1          1  ...       0         0        0               0         0   
2          1  ...       1         0        0               1         0   
3          1  ...       0         0        0               0         0   
4          1  ...       0         0        0               0         0   
..       ...  ...     ...       ...      ...             ...       ...   
370        1  ...       0         0        1               1         0   
371        1  ...       0         0        0               0         0   
372        1  ...       0         0        0               0         0   
373        1  ...       1         0        1               0         0   
374        0  ...       1         0        0               1         0   

     IRRITABILIDAD_MENOS5AÑOS  CORIZA  ANOSMIA  DISGEUSIA  OTROS  
0                           0       0        1          1      0  
1                           0       0        0          0      0  
2                           1       0        0          0      0  
3                           0       0        0          0      0  
4                           0       0        1          1      0  
..                        ...     ...      ...        ...    ...  
370                         0       0        0          0      0  
371                         0       0        0          0      0  
372                         0       0        0          0      0  
373                         1       0        1          0      0  
374                         0       0        0          0      0  

[375 rows x 29 columns]

In [37]:
# Reemplazamos los valores de las columnas 'RESULTADO PCR' y 'DESC_RESULT_PRUEBA_RAPIDA' por valores que faciliten
# las operaciones matriciales.
filtro_limpieza = selec_col.replace({'POSITIVO': '1', 'NEGATIVO':'0'},inplace=False)

# Manejamos a los datos "inconsistentes" de la columna 'RESULTADO PCR'.
filtro_limpieza['RESULTADO PCR'] = pd.to_numeric(filtro_limpieza['RESULTADO PCR'],errors='coerce')
filtro_limpieza = filtro_limpieza.replace(np.nan, -5, regex=True)

# Convertimos los valores de las columnas antes mencionadas a valores enteros.
filtro_limpieza['RESULTADO PCR'] = filtro_limpieza['RESULTADO PCR'].astype(int)
filtro_limpieza['DESC_RESULT_PRUEBA_RAPIDA'] = filtro_limpieza['DESC_RESULT_PRUEBA_RAPIDA'].astype(int)

filtro_limpieza

Num  EDD SEXO                                          MUNICIPIO  \
0      1   52    F  TLALPAN                                             
1      2   48    F  XOCHIMILCO                                          
2      3   40    F  XOCHIMILCO                                          
3      4   40    M  COYOACAN                                            
4      5   31    M  XOCHIMILCO                                          
..   ...  ...  ...                                                ...   
370  371   42    M  VENUSTIANO CARRANZA                                 
371  372   29    M  NICOLAS ROMERO                                      
372  373   38    F  COYOACAN                                            
373  374   45    F  XOCHIMILCO                                          
374  375   40    F  XOCHIMILCO                                          

     RESULTADO PCR  DESC_RESULT_PRUEBA_RAPIDA  INICIO_SUBITO  FIEBRE  TOS  \
0                1                          0              0       0    1   
1                1                          1              0       1    1   
2                1                          0              1       1    1   
3                0                          0              0       1    1   
4                0                          0              0       0    1   
..             ...                        ...            ...     ...  ...   
370              0                          0              0       0    1   
371              0                          0              1       1    1   
372              0                          0              1       0    1   
373             -5                          0              0       1    1   
374              1                          0              0       0    1   

     CEFALEA  ...  DISNEA  CIANOSIS  DIARREA  DOLOR_TORACICO  POLIPNEA  \
0          1  ...       0         0        0               0         0   
1          1  ...       0         0        0               0         0   
2          1  ...       1         0        0               1         0   
3          1  ...       0         0        0               0         0   
4          1  ...       0         0        0               0         0   
..       ...  ...     ...       ...      ...             ...       ...   
370        1  ...       0         0        1               1         0   
371        1  ...       0         0        0               0         0   
372        1  ...       0         0        0               0         0   
373        1  ...       1         0        1               0         0   
374        0  ...       1         0        0               1         0   

     IRRITABILIDAD_MENOS5AÑOS  CORIZA  ANOSMIA  DISGEUSIA  OTROS  
0                           0       0        1          1      0  
1                           0       0        0          0      0  
2                           1       0        0          0      0  
3                           0       0        0          0      0  
4                           0       0        1          1      0  
..                        ...     ...      ...        ...    ...  
370                         0       0        0          0      0  
371                         0       0        0          0      0  
372                         0       0        0          0      0  
373                         1       0        1          0      0  
374                         0       0        0          0      0  

[375 rows x 29 columns]

---
## 1. Análisis preeliminar de los datos.

In [59]:
# Hacemos un conteo de los datos consistentes e inconsistentes para los objetivos de la tarea.
datos_consistentes = filtro_limpieza[filtro_limpieza['RESULTADO PCR'] >= 0]
datos_inconsistentes = filtro_limpieza[filtro_limpieza['RESULTADO PCR'] < 0]
print(f'\nTotal de registros: {filtro_limpieza.shape[0]} \nNúmero de datos consistentes: {datos_consistentes.shape[0]} \
      \nNúmero de datos inconsistentess: {datos_inconsistentes.shape[0]} \n')


Total de registros: 375 
Número de datos consistentes: 362       
Número de datos inconsistentess: 13 



---
## 2. Análisis de la Concordancia.

In [62]:
# Extraemos los valores de las columnas 'RESULTADO PCR' y 'DESC_RESULT_PRUEBA_RAPIDA' en arreglos de NumPy para
# poder realizar operaciones matriciales de manera más eficiente.
pcr = datos_consistentes["RESULTADO PCR"].values
res_pcr = pcr.reshape(1,-1)

prueba_rapida = datos_consistentes["DESC_RESULT_PRUEBA_RAPIDA"].values
res_p_r = prueba_rapida.reshape(1,-1)

print(f'\nForma de Res_PCR: {res_pcr.shape} \n\nRes_PCR:\n {res_pcr}')
print(f'\n\nForma de Res_prueba_rapida: {res_p_r.shape} \n\nRes_prueba_rápida:\n {res_p_r} \n')


Forma de Res_PCR: (1, 362) 

Res_PCR:
 [[1 1 1 0 0 0 1 0 1 0 1 0 0 0 1 1 0 0 0 0 1 1 1 1 0 0 0 1 0 1 1 1 0 1 1 0
  1 1 1 1 1 0 0 0 1 1 0 0 0 0 1 1 0 1 1 0 0 0 0 0 0 1 0 0 1 0 1 0 1 0 0 0
  0 0 0 1 0 1 0 0 0 0 0 0 1 0 0 1 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
  0 0 0 0 0 0 0 0 0 0 1 1 1 0 1 1 1 0 1 1 0 0 0 1 1 1 0 0 0 0 0 1 0 0 1 1
  0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 1 1 0 0 0 1 0 0 0 1 0 1 0 0 0 0 1
  0 0 1 0 0 1 0 0 1 1 0 0 1 1 1 0 1 0 1 0 1 0 0 1 0 0 1 0 0 1 1 0 0 0 0 1
  0 0 0 0 0 0 1 0 0 1 0 0 1 0 0 0 1 0 1 0 0 0 1 0 1 0 0 0 0 1 0 1 0 1 0 1
  1 1 0 0 0 0 0 0 1 0 0 0 1 0 0 0 0 1 0 0 1 0 0 0 0 0 0 0 1 1 0 0 0 0 1 0
  0 0 1 0 0 0 0 1 0 0 1 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 0 1 0 0 0 0 0 1 0 0
  0 0 0 0 1 0 0 0 0 1 0 0 1 1 1 1 0 0 0 1 0 0 0 1 1 0 1 0 1 0 0 1 1 1 0 0
  0 1]]


Forma de Res_prueba_rapida: (1, 362) 

Res_prueba_rápida:
 [[0 1 0 0 0 0 0 0 0 0 1 0 1 0 1 0 0 0 0 1 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0
  1 0 0 1 0 0 0 0 0 1 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0 0
  0

* **Tabla general para el análisis de la concordancia**:

| PCR \ PR     | Positivo    | Negativo     | Total       |
| ------------ | :---------: | :----------: | :---------: |
| **Positivo** | a           | b            | _u_         |
| **Negativo** | c           | d            | _v_         |
| **Total**    | _r_         | _s_          | **N**       |

Donde:
* $u = a+b$
* $v = c+d$
* $r = a+c$
* $s = b+d$

In [40]:
# Realizamos operaciones matriciales para obtener la matriz de confusión (en este caso, de concordancia).
q=(np.array([1,0]).reshape(2,1) == res_pcr).astype(np.int64)
l=(res_p_r.reshape(-1,1) == np.array([1, 0]).reshape(1,2)).astype(np.int64)

cm = np.matmul(q,l)
print(f'\nMatriz de concordancia: \n {cm} \n')


Matriz de concordancia: 
 [[ 33  86]
 [  8 235]] 



Si sustituímos los valores obtenidos de la matriz de concordancia en la tabla, tenemos que:

| PCR \ PR     | Positivo    | Negativo     | Total     |
| ------------ | :---------: | :----------: | --------: |
| **Positivo** | 33          | 86           | _119_     |
| **Negativo** | 8           | 235          | _243_     |
| **Total**    | _41_        | _321_        | **362**   |

In [73]:
# Realizamos sumas sobre los renglones y sobre las columnas de la matriz de concordancia para obtener
# los valores de 'N','r', 's', 'u' y 'v'.
n = np.sum(cm)

suma_reng = np.sum(concordancia, axis=0)
r = suma_reng[0]
s = suma_reng[1]

suma_col = np.sum(concordancia,axis=1)
u = suma_col[0]
v = suma_col[1]

print(f'\nN: {n} \nr: {r} \t s: {s} \nu: {u} \t v:{v}')


N: 362 
r: 41 	 s: 321 
u: 119 	 v:243


Podemos comprobar estos resultados haciendo la búsqueda y conteo de cada valor de la tabla.

In [74]:
# Buscamos los registros que cumplan con la condición y contamos el número de los mismos.
pos_pos = datos_consistentes[(datos_consistentes["RESULTADO PCR"] == 1) & (datos_consistentes["DESC_RESULT_PRUEBA_RAPIDA"] == 1)]
pos_neg = datos_consistentes[(datos_consistentes["RESULTADO PCR"] == 1) & (datos_consistentes["DESC_RESULT_PRUEBA_RAPIDA"] == 0)]
neg_pos = datos_consistentes[(datos_consistentes["RESULTADO PCR"] == 0) & (datos_consistentes["DESC_RESULT_PRUEBA_RAPIDA"] == 1)]
neg_neg = datos_consistentes[(datos_consistentes["RESULTADO PCR"] == 0) & (datos_consistentes["DESC_RESULT_PRUEBA_RAPIDA"] == 0)]

print(f'\nMedición de concordancia: \n\n\t Número de observaciones en las cuales… \n \
      \n\t --------------------------------------------------------- \
      \n\t | Caso  | Prueba PCR \t | Prueba Rápida \t | Total |\
      \n\t --------------------------------------------------------- \
      \n\t | 1 \t | POSITIVO \t | POSITIVO \t\t | {pos_pos.shape[0]} \t |\
      \n\t | 2 \t | POSITIVO \t | NEGATIVO \t\t | {pos_neg.shape[0]} \t |\
      \n\t | 3 \t | NEGATIVO \t | POSITIVO \t\t | {neg_pos.shape[0]} \t |\
      \n\t | 4 \t | NEGATIVO \t | NEGATIVO \t\t | {neg_neg.shape[0]} \t |\
      \n\t ---------------------------------------------------------')


Medición de concordancia: 

	 Número de observaciones en las cuales… 
       
	 ---------------------------------------------------------       
	 | Caso  | Prueba PCR 	 | Prueba Rápida 	 | Total |      
	 ---------------------------------------------------------       
	 | 1 	 | POSITIVO 	 | POSITIVO 		 | 33 	 |      
	 | 2 	 | POSITIVO 	 | NEGATIVO 		 | 86 	 |      
	 | 3 	 | NEGATIVO 	 | POSITIVO 		 | 8 	 |      
	 | 4 	 | NEGATIVO 	 | NEGATIVO 		 | 235 	 |      
	 ---------------------------------------------------------


---
## 3. Obtención del porcentaje de concordancia.

La fórmula del **porcentaje de concordancia** es la siguiente:

$$
PC = \frac{a+d}{N} x 100
$$

In [75]:
# Obtención del porcentaje de concordancia ocupando la fórmula.
porcentaje_concordancia = ((cm[0][0]+cm[1][1])/n)*100
print(f'\nEl porcentaje de concordancia es: {round(porcentaje_concordancia,3)}% \n')


El porcentaje de concordancia es: 74.033% 



---
## 4. Obtención del índice de Kappa.

La fórmula del **índice de Kappa ($\kappa$)** es la siguiente:

$$
\kappa = \frac{C_o-C_e}{1-C_e}
$$
Donde:
* $C_o$: proporción de concordancias observadas.
* $C_e$: concordancia debida al azar.

Sabemos que:
$$
C_o = \frac{a+d}{N}
$$

In [76]:
# Cálculo de la proporción de concordancias observadas
c_o = (cm[0][0]+cm[1][1])/n
print(f'\nEl valor de la proporción de concordancias observadas Co es: {c_o} \n')


El valor de la proporción de concordancias observadas Co es: 0.7403314917127072 



Por otro lado:
$$
C_e = \frac{a_e+d_e}{N}
$$

* Donde:

$$a_e = \frac{r*u}{N}$$ 

$$d_e = \frac{s*v}{N}$$

In [80]:
# Cálculo de los valores esprados de las celdas 'a' y 'd' de la tabla de concordancia.
a_e = (r*u)/n
d_e = (s*v)/n
print(f'\nA_e = {a_e} \nD_e = {d_e} \n')


A_e = 13.477900552486188 
D_e = 215.47790055248618 



Si sustituímos los valores antes obtenidos, tenemos que:

In [81]:
# Cálculo de la concordancia debida al azar.
c_e = (a_e+d_e)/n
print(f'\nEl valor de la concordancia debida al azar es: {c_e} \n')


El valor de la concordancia debida al azar es: 0.632474588687769



Ya que tenemos todos los valores necesarios, podemos utilizar la fórmula de **$\kappa$**.

In [82]:
# Cálculo del índice de Kappa.
k = (c_o-c_e)/(1-c_e)
print(f'\nEl índice de kappa es: {k} \n')


El índice de kappa es: 0.29346787924089535 



---
## 5. Obtención de los valores predictivos

Veamos las fórmulas que vamos a ocupar:

* **Valor Predictivo Positivo**
$$
VPP = \frac{VP}{VP+FP}
$$


* **Valor Predictivo Negativo**
$$
VPN = \frac{VN}{VN+FN}
$$

Volvamos a echar un ojo a la **tabla para el análisis de la concordancia**:

| PCR \ PR     | Positivo    | Negativo     |
| ------------ | :---------: | :----------: |
| **Positivo** | 33          | 86           |
| **Negativo** | 8           | 235          |

¿Cuáles son los verdaderos y falsos **positivos**? ¿Y de los **negativos**?

In [85]:
# Obtención de los valores predictivos.
vp = cm[0][0]
fp = cm[1][0]
vn = cm[1][1]
fn = cm[0][1]

vpp = vp/(vp+fp)
vpn = vn/(fn+vn)
print(f'\nValor Predictivo Positivo: {round(vpp*100,3)}% \nValor Predictivo Negativo: {round(vpn*100,3)}% \n')


Valor Predictivo Positivo: 80.488% 
Valor Predictivo Negativo: 73.209% 



---
## 6. Análisis de los datos clínicos de px positivos a COVID.

In [87]:
# Filtramos a los pacientes que dieron positivo a COVID.
pos_pos = datos_consistentes[datos_consistentes["RESULTADO PCR"] == 1]

# Filtramos las columnas de mayor relevancia para el inciso.
pos_pos = pos_pos[['INICIO_SUBITO', 'FIEBRE', 'TOS', 'CEFALEA', 'ODINOFAGIA', 
                                   'ATAQUE_AL_ESTADO_GENERAL', 'MIALGIAS', 'ARTRALGIAS', 'POSTRACION', 
                                   'RINORREA', 'ESCALOFRIO', 'DOLOR_ABDOMINAL', 'CONJUNTIVITIS', 'DISNEA', 
                                   'CIANOSIS', 'DIARREA', 'DOLOR_TORACICO', 'POLIPNEA', 'IRRITABILIDAD_MENOS5AÑOS', 
                                   'CORIZA', 'ANOSMIA', 'DISGEUSIA', 'OTROS']]

pos_pos

INICIO_SUBITO  FIEBRE  TOS  CEFALEA  ODINOFAGIA  \
0                0       0    1        1           1   
1                0       1    1        1           1   
2                1       1    1        1           1   
6                1       1    1        0           1   
8                0       1    0        1           1   
..             ...     ...  ...      ...         ...   
363              0       1    1        1           1   
366              0       1    1        1           0   
367              0       0    1        0           0   
368              0       1    1        0           0   
374              0       0    1        0           1   

     ATAQUE_AL_ESTADO_GENERAL  MIALGIAS  ARTRALGIAS  POSTRACION  RINORREA  \
0                           1         1           1           0         1   
1                           1         1           1           0         0   
2                           1         1           1           0         0   
6                           0         0           0           0         0   
8                           1         1           1           0         0   
..                        ...       ...         ...         ...       ...   
363                         0         0           0           0         1   
366                         0         0           0           0         0   
367                         0         0           0           0         0   
368                         1         0           0           0         0   
374                         1         0           0           0         1   

     ...  DISNEA  CIANOSIS  DIARREA  DOLOR_TORACICO  POLIPNEA  \
0    ...       0         0        0               0         0   
1    ...       0         0        0               0         0   
2    ...       1         0        0               1         0   
6    ...       0         0        0               0         0   
8    ...       0         0        0               0         0   
..   ...     ...       ...      ...             ...       ...   
363  ...       0         0        0               0         0   
366  ...       0         0        0               0         0   
367  ...       0         0        0               0         0   
368  ...       0         0        1               1         0   
374  ...       1         0        0               1         0   

     IRRITABILIDAD_MENOS5AÑOS  CORIZA  ANOSMIA  DISGEUSIA  OTROS  
0                           0       0        1          1      0  
1                           0       0        0          0      0  
2                           1       0        0          0      0  
6                           0       0        1          0      0  
8                           0       0        0          0      0  
..                        ...     ...      ...        ...    ...  
363                         0       0        0          0      0  
366                         0       0        0          0      0  
367                         0       0        0          0      0  
368                         1       0        0          0      0  
374                         0       0        0          0      0  

[119 rows x 23 columns]

La siguiente tabla nos muestra el _ranking_ de los síntomas más comunes que tuvieron los pacientes positivos a COVID (basándonos **solamente** en el resultado de la PCR).

In [88]:
# Conteo de los síntomas de los pacientes positivos a COVID.
sum_datos_col = np.sum(pos_pos, axis=0)

# Creación de un DataFrame para una mejor visualización.
d_c = pd.DataFrame(sum_datos_col)
d_c = d_c.rename(columns = {0:'Conteo'})
datos_clinicos = d_c.sort_values(by='Conteo',ascending=False)
datos_clinicos

Conteo
TOS                          105
CEFALEA                       87
FIEBRE                        74
ODINOFAGIA                    64
ATAQUE_AL_ESTADO_GENERAL      54
MIALGIAS                      54
RINORREA                      53
ARTRALGIAS                    46
ESCALOFRIO                    37
DISGEUSIA                     32
ANOSMIA                       31
INICIO_SUBITO                 28
DOLOR_TORACICO                24
DIARREA                       23
DISNEA                        23
DOLOR_ABDOMINAL               19
POSTRACION                     7
IRRITABILIDAD_MENOS5AÑOS       6
CORIZA                         6
CONJUNTIVITIS                  5
CIANOSIS                       1
POLIPNEA                       1
OTROS                          0

---
## 7. Px asintomáticos positivos a COVID.

Vamos a obtener a los pacientes asintomáticos buscando aquellos registros en los que su suma de síntomas sea igual a 0.

In [90]:
# Suma de síntomas por registro de paciente.
sum_datos_reng = np.sum(pos_pos, axis=1)

# Obtención de los índices de los pacientes asintomáticos.
lista_asintomaticos = conteo_sintomas.index[conteo_sintomas['Conteo'] == 0].tolist()
print(f'Lista de los índices de pacientes asintomáticos: {lista_asintomaticos}')

Lista de los pacientes asintomáticos: [272, 277]


In [91]:
# Obtención de los datos más relevantes de los pacientes asintomáticos en el DataFrame original.
asin = df.iloc[lista_asintomaticos]
asin = asintomaticos  = asin[['Num', 'EDD', 'SEXO', 'MUNICIPIO',
       'FECHA_ATENCION_INGRESO', 'FECHA_INICIO_CUADRO_CLINICO',
       'RESULTADO PCR', 'DESC_RESULT_PRUEBA_RAPIDA',
       'MOTIVO DE EGRESO']]
asintomaticos

Num  EDD SEXO MUNICIPIO FECHA_ATENCION_INGRESO  \
272  273   47    M       NaN             2021-08-27   
277  278   30    M       NaN             2021-08-30   

    FECHA_INICIO_CUADRO_CLINICO RESULTADO PCR DESC_RESULT_PRUEBA_RAPIDA  \
272                  2021-08-21      POSITIVO                  NEGATIVO   
277                  2021-08-16      POSITIVO                  NEGATIVO   

    MOTIVO DE EGRESO  
272          MEJORIA  
277          MEJORIA

* Es interesante notar que para ambos pacientes su Prueba rápida tuvo un resultado negativo. 

---